lib/cc/engine/analyzers/reporter.rb

```diff
@@ -0,0 +1,67 @@
+require 'cc/engine/analyzers/violation'
+
+module CC
+  module Engine
+    module Analyzers
+      class Reporter
+        def initialize(directory, language_strategy, io)
+          @directory = directory
+          @language_strategy = language_strategy
+          @io = io
+          @flay = Flay.new(flay_options)
+        end
+
+        def run
+          sexps = language_strategy.run
+          sexps.each do |sexp|
+            process_sexp(sexp)
+          end
+
+          report
+        end
+
+        def report
+          flay.report(StringIO.new).each do |issue|
```

Comment:
what's the significance of `StringIo.new` here? could this get moved out to a well-named method signifying what it's used for?